src/test/java/com/jcabi/jdbc/JdbcSessionITCase.java

```diff
@@ -91,6 +99,93 @@ public void changesTransactionIsolationLevel() throws Exception {
+                " day OUT date)",
+                " AS $$ BEGIN SELECT name, CURRENT_DATE INTO username, day",
+                " FROM  users; END; $$ LANGUAGE plpgsql;"
+            )
+        ).execute().commit();
+        final Object[] result = new JdbcSession(source)
+            .sql("{call fetchUser(?, ?)}")
+            .prepare(
+                new Preparation() {
+                    @Override
+                    public void
+                        prepare(final PreparedStatement stmt)
+                        throws SQLException {
+                            final CallableStatement cstmt =
+                                (CallableStatement) stmt;
+                            cstmt.registerOutParameter(1, Types.VARCHAR);
+                            cstmt.registerOutParameter(2, Types.DATE);
+                    }
+                }
+             )
+            .call(new StoredProcedureOutcome<Object[]>(2));
+        MatcherAssert.assertThat(result.length, Matchers.is(2));
+        MatcherAssert.assertThat(
+            result[0].toString(),
+            Matchers.containsString("Charles")
+        );
+        MatcherAssert.assertThat(
+            (Date) result[1],
+            Matchers.notNullValue()
+        );
+    }
+
+    /**
+     * JdbcSession can run a function (stored procedure) with
+     * input and output parameters.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void callsFunctionWithInOutParam() throws Exception {
+        final DataSource source = JdbcSessionITCase.source();
+        new JdbcSession(source).autocommit(false).sql(
+            "CREATE TABLE IF NOT EXISTS usersid (id INTEGER, name VARCHAR(50))"
+        ).execute().sql("INSERT INTO usersid (id, name) VALUES (?, ?)")
+        .set(1).set("Marco Polo").execute().sql(
+            StringUtils.join(
+                "CREATE OR REPLACE FUNCTION fetchUserById(uid IN INTEGER,",
+                " usrnm OUT text) AS $$ BEGIN",
+                " SELECT name INTO usrnm FROM  usersid WHERE id=uid;",
+                " END; $$ LANGUAGE plpgsql;"
+            )
+        ).execute().commit();
+        final Object[] result = new JdbcSession(source)
+            .sql("{call fetchUserById(?, ?)}")
+            .set(1)
+            .prepare(
+                new Preparation() {
+                    @Override
+                    public void
+                        prepare(final PreparedStatement stmt)
+                        throws SQLException {
+                            ((CallableStatement) stmt)
+                                .registerOutParameter(2, Types.VARCHAR);
+                    }
+                }
+             )
+            .call(new StoredProcedureOutcome<Object[]>(new int[] {2}));
+        MatcherAssert.assertThat(result.length, Matchers.is(1));
+        MatcherAssert.assertThat(
+            result[0].toString(),
+            Matchers.containsString("Polo")
```

Comment:
@mkordas because then it complains of  douvle literals and it is safe enough to have a contains, rahter than declare a variable for it... all the tests in all the jcabi libs are like this. Contains is used instead of equals wjere appropriate